clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -90,7 +90,7 @@
   private final LifeCycleMessageFactory messageFactory;
   private final Map<Class<? extends EhcacheEntityResponse>, List<ResponseListener<? extends EhcacheEntityResponse>>> responseListeners = new ConcurrentHashMap<Class<? extends EhcacheEntityResponse>, List<ResponseListener<? extends EhcacheEntityResponse>>>();
   private final List<DisconnectionListener> disconnectionListeners = new CopyOnWriteArrayList<DisconnectionListener>();
-  private final List<ReconnectListener> reconnectListeners = new ArrayList<ReconnectListener>();
+  private final List<ReconnectListener> reconnectListeners = new CopyOnWriteArrayList<ReconnectListener>();
```

Comment:
Discussed with @AbfrmBlr and the reason it was done with synchronised was not just for visibility but to make every access to that list exclusive. Even iteration over that list has to be exclusive with other mutations. This exclusiveness is required to avoid listener registration/deregistartion races while a reconnect is happening (iteration over the listeners). So reverting this change.